packages/flutter/lib/src/rendering/sliver_padding.dart

```diff
@@ -92,38 +92,34 @@ class RenderSliverPadding extends RenderSliver with RenderObjectWithChildMixin<R
     return null;
   }
 
-  /// The padding in the cross-axis direction on the side to the right when
-  /// facing away from the zero scroll offset in the scroll axis direction. (In
-  /// other words, for a vertical downwards-growing list, the padding on the
+  /// The padding in the cross-axis direction on the side closest to the origin.
+  /// (In other words, for a vertical downwards-growing list, the padding on the
   /// left.)
   ///
   /// Only valid after layout has started, since before layout the render object
   /// doesn't know what direction it will be laid out in.
-  double get startPadding {
+  double get leadingPadding {
```

Comment:
The bug is very simple:

1) Creating a scroller in the up direction.
2) Put a blue box in the scroller.
3) Wrap the blue box in a SliverPadding with const EdgeDims.only(left: 10.0).

Do you want the blue box to touch the left or the right edge of the scroller?  Today, it touches the left edge because the 10.0 of "left" padding actually shows up on screen on the right.